### PR TITLE
feat: Abort controller in Config maps, Secrets and Deployment template

### LINKED ISSUE #1360: Cancel token implementation in Deployment template, ConfigMap and Secrets
**Please Describe The Problem To Be Solved**
Added cancellation token in deployment template and Included Abort Controller for ConfigMaps and Secrets to abort API call if there are multiple API calls from UI when data of the API hasn’t fetched. Abort Controller included for Protected Configuration also.

**(Optional): Suggest A Solution**
(Replace This Text: A concise description of your preferred solution. Things to address include:
* Details of the technical implementation
* Tradeoffs made in design decisions
* Caveats and considerations for the future

If there are multiple solutions, please present each one separately. Save comparisons for the very end.)

### PR DESCRIPTION
**Pull request by dhananjay0601**
# Description

Included Abort Controller for ConfigMaps and Secrets to abort API call if there are multiple API calls from UI when data of the API hasn’t fetched. Abort Controller included for Protected Configuration also.

Fixes #1360

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] This has been tested by sending multiple API calls with slow network connections where the previous API call has been aborted if the data hasn’t fetched yet.

# Checklist:

* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [X] I have performed a self-review of my own code
* [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/CIPipelineN/CustomInputVariableSelect.tsx
Comment (vivek-devtron) [inline src/components/CIPipelineN/CustomInputVariableSelect.tsx:1]: Please revert this change

File: src/components/ConfigMapSecret/ConfigMapSecret.components.tsx [hunk 1]
```diff
@@ -191,14 +195,19 @@ export function ConfigMapSecretContainer({
...
+            }
             if (
                 (_cmSecretData?.status === 'rejected' && _cmSecretData?.reason?.code === 403) ||
                 (_draftData?.status === 'rejected' && _draftData?.reason?.code === 403)
             ) {
                 toast.warn(<ToastBody title="View-only access" subtitle="You won't be able to make any changes" />)
             }
         } catch (error) {
+            if(error === 'AbortError') {
+                console.log("Request was aborted", error)
```
Comment (vivek-devtron) [inline src/components/ConfigMapSecret/ConfigMapSecret.components.tsx:209]: In case of abort there should be no error message

File: src/components/ConfigMapSecret/service.ts [hunk 2]
```diff
@@ -27,11 +27,11 @@ export function overRideConfigMap(id, appId, environmentId, configData) {
     })
 }
 
-export function getConfigMapList(appId, envId?) {
+export function getConfigMapList(appId, envId?, signal?) {
```
Comment (vivek-devtron) [inline src/components/ConfigMapSecret/service.ts:30]: Why signal is an optional param?

File: yarn.lock
Comment (vivek-devtron) [inline yarn.lock:1]: revert this file change

File: src/components/common/helpers/Helpers.tsx [hunk 3]
```diff
@@ -944,7 +944,7 @@ export const convertToOptionsList = (
     })
 }
 
-export const importComponentFromFELibrary = (componentName: string, defaultComponent?, type?: string) => {
+export const importComponentFromFELibrary = (componentName: string, defaultComponent?, type?: string, signal?: string) => {
```
Comment (vivek-devtron) [inline src/components/common/helpers/Helpers.tsx:947]: why are we passing signal here?

File: yarn.lock
Comment (vivek-devtron) [inline yarn.lock:1]: Please revert this file

File: src/components/ConfigMapSecret/ConfigMapSecret.components.tsx [hunk 4]
```diff
@@ -119,6 +119,7 @@ export function ConfigMapSecretContainer({
     const [isLoader, setLoader] = useState<boolean>(false)
     const [draftData, setDraftData] = useState(null)
     const [selectedTab, setSelectedTab] = useState(data?.draftState === 4 ? 2 : 3)
+    const [abortController, setAbortController] = useState(new AbortController());
```
Comment (shivani170) [inline src/components/ConfigMapSecret/ConfigMapSecret.components.tsx:122]: Is there any need to store controller locally?

**Review by vivek-devtron — APPROVED** [review]
